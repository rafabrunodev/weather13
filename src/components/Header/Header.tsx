import styled from "styled-components";
import { colors, device } from "../../utils";

const HeaderEl = styled.header`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 20px 40px;
  color: ${colors.main};
  flex-direction: column;

  @media ${device.laptop} {
    flex-direction: row;
    padding-top: 0;
  }
`;

const Title = styled.h2`
  font-size: 30px;
  margin: 10px;

  @media ${device.laptop} {
    padding: 30px;
    margin-left: -40px;
  }
`;

const Link = styled.a`
  color: ${colors.main};

  &:hover{
    color: ${colors.main};
    opacity: .8;
  }
`;

const Header: React.FC = () => {
  return (
    <HeaderEl>
      <Link href="https://www.linkedin.com/in/rafaelbrunoweb/">
        by Rafael Bruno
      </Link>
      <Title>Weather13</Title>
      <Link href="">GitHub</Link>
    </HeaderEl>
  );
};

export default Header;
