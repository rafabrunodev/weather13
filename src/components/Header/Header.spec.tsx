import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import Header from "./Header";

describe("Header.tsx", () => {
  it("should match Snapshot", () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
