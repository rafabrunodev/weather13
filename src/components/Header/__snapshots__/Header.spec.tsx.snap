// Vitest Snapshot v1

exports[`Header.tsx > should match Snapshot 1`] = `
<div>
  <header
    class="sc-bcXHqe fDtTUV"
  >
    <a
      class="sc-dkrFOg bdQghR"
      href="https://www.linkedin.com/in/rafaelbrunoweb/"
    >
      by Rafael Bruno
    </a>
    <h2
      class="sc-gswNZR bvYKxX"
    >
      Weather13
    </h2>
    <a
      class="sc-dkrFOg bdQghR"
      href=""
    >
      GitHub
    </a>
  </header>
</div>
`;
